### PR TITLE
feat: add C implementation for `stats/base/dists/erlang/cdf`

### LINKED ISSUE #3568: [RFC]: Add C implementation for `@stdlib/stats/base/dists/erlang/cdf`
### Description

This RFC proposes adding a C implementation, including a Node.js native addon, C benchmarks, and C examples, for the [`@stdlib/stats/base/dists/erlang/cdf`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/@stdlib/stats/base/dists/erlang/cdf) package.

When adding support, the following tasks should be completed:

1. A C source implementation should be added to a `src` folder within the package's directory.
2. A native add-on interface should be added in order to call the C implementation from JavaScript.
3. A JavaScript file should be added to the `lib` folder to allow benchmarking and unit testing the native implementation from JavaScript.
4. A new test file should be created which includes unit tests testing against expected behavior.
5. C benchmarks should be added which measure performance for the C source implementation.
6. A file containing C examples should be added to demonstrate example usage.
7. JavaScript benchmarks should be added which measure performance for the native add-on.
8. Build and configuration files should be added to allow compiling and running the C source implementation using project `make` commands (as documented below).
9. The README should be updated to include documentation for the added C API.

To provide a concrete example of what a PR adding the desired method should contain, see https://github.com/stdlib-js/stdlib/pull/3354, which is a PR adding a C implementation for the CDF of an arcsine distribution. This should provide an idea of what is expected.

#### Prerequisites

1. Review JavaScript implementation in `lib/main.js` and its required modules to identify all `stdlib` functions used.
2. Verify C implementations exist for each required function.

### Related Issues

#3355

### Questions

No.

### Other

Once the implementation is ready including C examples and benchmarks, we should be able to run the following `make` commands.

#### Build native add-on

```bash
NODE_ADDONS_PATTERN="@stdlib/stats/base/dists/erlang/cdf" make install-node-addons
```

#### Run C examples

```bash
make examples-c-files FILES="$(pwd)/lib/node_modules/@stdlib/stats/base/dists/erlang/cdf/examples/c/example.c"
```

#### Run C benchmarks

```bash
make benchmark-c-files FILES="$(pwd)/lib/node_modules/@stdlib/stats/base/dists/erlang/cdf/benchmark/c/benchmark.c"
```

#### Run JavaScript benchmarks

```bash
make benchmark-javascript-files FILES="$(pwd)/lib/node_modules/@stdlib/stats/base/dists/erlang/cdf/benchmark/benchmark.native.js"
```

#### Run all tests

```bash
make TESTS_FILTER=".*/@stdlib/stats/base/dists/erlang/cdf/.*" test
```

### Notes

- If you are interested in contributing a PR which addresses this RFC and are a **first-time contributor** or not yet familiar with our project conventions, please minimize the submission of LLM-generated code. Please consult our [contributing guidelines](https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md) and the associated [development guide](https://github.com/stdlib-js/stdlib/blob/develop/docs/development.md). Failure to respect project conventions will result in your PR being rejected without review. Thank you for understanding!

### Checklist

- [x] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [x] Searched for existing issues and pull requests.
- [x] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by Lokeshranjan8**
Resolves #3568 .

## Description

> What is the purpose of this pull request?

This pull request:

-   `adds C implementation for stats/base/dists/erlang/cdf along with relevant docs, tests, examples and benchmarks`

## Related Issues

> Does this pull request have any related issues?

This pull request has the following related issues:

-   

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

### AI Assistance

> When authoring the changes proposed in this PR, did you use any kind of AI assistance?

-   [ ] Yes
-   [x] No

If you answered "yes" above, how did you use AI assistance?

-   [ ] Code generation (e.g., when writing an implementation or fixing a bug)
-   [ ] Test/benchmark generation
-   [ ] Documentation (including examples)
-   [ ] Research and understanding

#### Disclosure

> If you answered "yes" to using AI assistance, please provide a short disclosure indicating how you used AI assistance. This helps reviewers determine how much scrutiny to apply when reviewing your contribution. Example disclosures: "This PR was written primarily by Claude Code." or "I consulted ChatGPT to understand the codebase, but the proposed changes were fully authored manually by myself.".

{{TODO: add disclosure if applicable}}

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/stats/base/dists/erlang/cdf/benchmark/c/benchmark.c [hunk 1]
```diff
@@ -89,40 +95,30 @@ static float rand_float( void ) {
...
 */
 static double benchmark( void ) {
-	float complex z1[ 100 ];
-	float complex z2[ 100 ];
-	float complex z3[ 100 ];
-	float complex z4;
+	double x[ 100 ];
+	int32_t k[ 100 ];
+	double lambda[ 100 ];
 	double elapsed;
```
Comment (Neerajpathak07) [inline lib/node_modules/@stdlib/stats/base/dists/erlang/cdf/benchmark/c/benchmark.c:101]:
```suggestion
static double benchmark( void ) {
	double lambda[ 100 ];
        int32_t k[ 100 ];
	double x[ 100 ];
	double elapsed;
```
ordering by length

File: lib/node_modules/@stdlib/stats/base/dists/erlang/cdf/src/main.c [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+* @param x        input value
+* @param k        shape parameter
+* @param lambda   rate parameter
+* @returns        evaluated CDF
+*
+* @example
+* double y = stdlib_base_dists_erlang_cdf( 2.0, 1, 1.0 );
+* // returns ~0.865
+*/
+double stdlib_base_dists_erlang_cdf( const double x, const double k, const double lambda ) {
```
Comment (Neerajpathak07) [inline lib/node_modules/@stdlib/stats/base/dists/erlang/cdf/src/main.c]: I'm a little confused here, since in `main.c` you are utlizing `double`  dtype for `k` would be better to streamline the C impementation to use `int32_t`.

File: lib/node_modules/@stdlib/stats/base/dists/erlang/cdf/src/addon.c [hunk 3]
```diff
@@ -16,7 +16,7 @@
 * limitations under the License.
 */
 
-#include "stdlib/complex/float32/base/add3.h"
+#include "stdlib/stats/base/dists/erlang/cdf.h"
 #include "stdlib/math/base/napi/ternary.h"
 
-STDLIB_MATH_BASE_NAPI_MODULE_CCC_C( stdlib_base_complex64_add3 )
+STDLIB_MATH_BASE_NAPI_MODULE_DDD_D( stdlib_base_dists_erlang_cdf )
```
Comment (Neerajpathak07) [inline lib/node_modules/@stdlib/stats/base/dists/erlang/cdf/src/addon.c]: Use `DID_D` macro here.

File: lib/node_modules/@stdlib/stats/base/dists/erlang/cdf/lib/native.js [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+var addon = require( './../src/addon.node' );
+
+
+// MAIN //
+
+/**
+* Evaluates the cumulative distribution function (CDF) for an Erlang distribution with shape parameter `k` and rate parameter `lambda` at a value `x`.
+*
+* @param {number} x - input value
```
Comment (Neerajpathak07) [inline lib/node_modules/@stdlib/stats/base/dists/erlang/cdf/lib/native.js:32]:
```suggestion
*
* @private
* @param {number} x - input value
```

File: lib/node_modules/@stdlib/stats/base/dists/erlang/cdf/test/test.native.js [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+	y = cdf( 2.0, NINF, NINF );
+	t.strictEqual( isnan( y ), true, 'returns expected value' );
+
+	y = cdf( 2.0, NaN, NINF );
+	t.strictEqual( isnan( y ), true, 'returns expected value' );
+
+	t.end();
+});
+
+tape( 'the function evaluates the cdf for `x` given large `k` and `lambda`', opts, function test( t ) {
```
Comment (Neerajpathak07) [inline lib/node_modules/@stdlib/stats/base/dists/erlang/cdf/test/test.native.js]: could you streamline this test cases to adhere to the latest code convention which we follow for statistical distribution packages.
you can refer to:- https://github.com/stdlib-js/stdlib/blob/d9a5e081c5376c8688ffd96f29dec0e135cb123c/lib/node_modules/%40stdlib/stats/base/dists/wald/pdf/test/test.pdf.js#L197

here you mainly have to update the upstream fixtures for the package to house them inside a single `data.json` file. After which it can be sourced to test various edge cases as per done in the ref above.

Another nitpick would be to use `isAlmostSameValue`.

**Review by Neerajpathak07 — CHANGES_REQUESTED** [review]
@Lokeshranjan8 Have left out an inital round of comments. Mainly you'll have to check your implementation and rectify which dtype `k` uses and which corresponding macro will be required for it.
To get a good idea of the current code convention for these pacakges take a look at:- https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/stats/base/dists/wald/pdf

File: lib/node_modules/@stdlib/stats/base/dists/erlang/cdf/src/main.c (hunk 2 above)
Comment (Neerajpathak07) [reply lib/node_modules/@stdlib/stats/base/dists/erlang/cdf/src/main.c]: you can also take a look at other `erlang/*` packages to get a clear perspective of the dtype.

**Comment by Lokeshranjan8** [comment]
@Neerajpathak07  Thanks for the help

File: lib/node_modules/@stdlib/stats/base/dists/erlang/cdf/benchmark/c/benchmark.c [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+	double k[ 100 ];
+	double x[ 100 ];
+	double elapsed;
+	double y;
+	double t;
+	int i;
+
+	for ( i = 0; i < 100; i++ ) {
+		x[ i ] = random_uniform( 0.0, 100.0 );
+		k[ i ] = random_uniform( 1.0, 100.0 );
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/erlang/cdf/benchmark/c/benchmark.c]: `k` must be a non-negative integer, but `random_uniform( 1.0, 100.0 )` produces fractional values, causing `stdlib_base_dists_erlang_cdf` to return NaN on the first iteration and the benchmark to immediately break.

File: lib/node_modules/@stdlib/stats/base/dists/erlang/cdf/src/main.c [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+#include "stdlib/math/base/assert/is_nonnegative_integer.h"
+#include "stdlib/stats/base/dists/gamma/cdf.h"
+
+/**
+* Evaluates the cumulative distribution function (CDF) for an Erlang distribution with shape parameter `k` and rate parameter `lambda` at a value `x`.
+*
+* @param x         input value
+* @param k         shape parameter
+* @param lambda    rate parameter
+* @returns         evaluated CDF
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/erlang/cdf/src/main.c]: Use `@return`, not `@returns`, to match the other C sources.

```suggestion
* @return          evaluated CDF
```

File: lib/node_modules/@stdlib/stats/base/dists/erlang/cdf/src/main.c [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+*
+* @example
+* double y = stdlib_base_dists_erlang_cdf( 2.0, 1, 1.0 );
+* // returns ~0.865
+*/
+double stdlib_base_dists_erlang_cdf( const double x, const double k, const double lambda ) {
+	if ( !stdlib_base_is_nonnegative_integer( k ) ) {
+		return 0.0 / 0.0; //NaN
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/erlang/cdf/src/main.c]: Space after `//`.

```suggestion
		return 0.0 / 0.0; // NaN
```

File: lib/node_modules/@stdlib/stats/base/dists/erlang/cdf/README.md [hunk 9]
```diff
@@ -164,6 +164,113 @@ logEachMap( 'x: %0.4f, k: %d, λ: %0.4f, F(x;k,λ): %0.4f', x, k, lambda, cdf );
...
+    double k;
+    double y;
+    int i;
+
+    for ( i = 0; i < 25; i++ ) {
+        x = random_uniform( 0.0, 10.0 );
+        k = stdlib_base_round( random_uniform( 0.0, 10.0 ) );
+        lambda = random_uniform( 0.0, 5.0 );
+        y = stdlib_base_dists_erlang_cdf( x, k, lambda );
+        printf( "x: %lf, k: %d, λ: %lf, F(x;k,λ): %lf\n", x, k, lambda, y );
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/erlang/cdf/README.md]: `k` is a `double`, so `%d` is undefined behavior — use `%lf`, as in `examples/c/example.c`.

```suggestion
        printf( "x: %lf, k: %lf, λ: %lf, F(x;k,λ): %lf\n", x, k, lambda, y );
```

File: lib/node_modules/@stdlib/stats/base/dists/erlang/cdf/benchmark/benchmark.native.js [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+var opts = {
+	'skip': ( cdf instanceof Error )
+};
+
+
+// MAIN //
+
+bench( format( '%s::native', pkg ), opts, function benchmark( b ) {
+	var lambda;
+	var opts;
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/erlang/cdf/benchmark/benchmark.native.js]: Drop the inner `opts` — it shadows the module-level `opts` and trips the linter. Pass the dtype object inline to the `discreteUniform`/`uniform` calls, as done in `gamma/cdf/benchmark/benchmark.native.js`.

**Review by Planeshifter — CHANGES_REQUESTED** [review]

File: lib/node_modules/@stdlib/stats/base/dists/erlang/cdf/benchmark/c/benchmark.c [hunk 11]
```diff
@@ -0,0 +1,145 @@
...
+	double k[ 100 ];
+	double x[ 100 ];
+	double elapsed;
+	double y;
+	double t;
+	int i;
+
+	for ( i = 0; i < 100; i++ ) {
+		x[ i ] = random_uniform( 0.0, 100.0 );
+		k[ i ] = stdlib_base_round( random_uniform( 0.0, 100.0 ) );
```
Comment (Neerajpathak07) [inline lib/node_modules/@stdlib/stats/base/dists/erlang/cdf/benchmark/c/benchmark.c:108]: updated this using `stdlib_base_round` similar to what `erlang/mgf` does for non-negative integer values of `k`

**Review by Planeshifter — APPROVED** [review]
Looks good with recent edits. Thanks, @Lokeshranjan8 and @Neerajpathak07!

**Comment by Neerajpathak07** [comment]
@Planeshifter Let's land this in!!!
